web/src/settings_users.js

```diff
@@ -150,23 +150,47 @@ function role_selected_handler(event, dropdown, widget) {
     widget.render();
 }
 
-function get_roles() {
+function get_roles(user_ids) {
+    return get_roles_with_counts(user_ids);
+}
+
+function count_users_by_role(user_ids) {
+    const role_counts = {};
+
+    for (const user_id of user_ids) {
+        const user = people.get_by_user_id(user_id);
+        const role_code = user.role;
+
+        if (!role_counts[role_code]) {
+            role_counts[role_code] = 0;
+        }
+        role_counts[role_code] += 1;
```

Comment:
Maybe we could simplify here with `role_counts[role_code] = (role_counts[role_code] || 0) + 1;` instead?

Reply:
I've made those changes as shown above to improve readability. Should I change it, or leave it as it is? What do you think exactly?